### DISCUSSION TITLE
Merge matrix build artifacts for release

### DISCUSSION THREAD

**Discussion by thebino** [community/community#25338] [Actions]
<p>I am using a matrix configuration to compile my project on different platforms and want to merge all the outputs to one single release.</p>
<pre><code class="lang-auto">jobs:
  build:
    name: ${{ matrix.config.name }}
    runs-on: ${{ matrix.config.os }}
</code></pre>
<p>I already uploaded all artifacts to the workflow, but I still need to put them into a new release</p>
<pre><code class="lang-auto">    - name: Upload artifacts
      uses: actions/upload-artifact@v2
      with:
        name: ${{ matrix.config.artifact }}
        path: build/app*
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/231468bdb2eab787f4e6c13d3ae18382a83bb6a9.png" data-download-href="/uploads/short-url/50kr97pTmwdhdj8lz5BceUMgwWt.png?dl=1" title="Screenshot 2020-06-15 at 19.17.55" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/841681/181090499-8bae5c12-136f-4590-a586-71c0d15646c6.png" alt="Screenshot 2020-06-15 at 19.17.55" data-base62-sha1="50kr97pTmwdhdj8lz5BceUMgwWt" width="690" height="471" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/231468bdb2eab787f4e6c13d3ae18382a83bb6a9_2_690x471.png, https://aws1.discourse-cdn.com/github/optimized/2X/2/231468bdb2eab787f4e6c13d3ae18382a83bb6a9_2_1035x706.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/2/231468bdb2eab787f4e6c13d3ae18382a83bb6a9.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/231468bdb2eab787f4e6c13d3ae18382a83bb6a9_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-06-15 at 19.17.55</span><span class="informations">1294×884 63.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by chrispat** [comment]
<p>You could create a job that depends on the jobs that build your artifacts <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idneeds" rel="nofollow noopener">https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idneeds</a>.  You can then have that job download all of the artifacts using <a href="https://github.com/actions/download-artifact" rel="nofollow noopener">https://github.com/actions/download-artifact</a> and upload those as release assets.</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/thebino">@thebino</a>,</p>
<p>You can create a new job which needs your matrix jobs, in the new job, you can</p>
<ol>
<li>Download the artifacts to one folder.</li>
<li>Create a new release with action ‘actions/create-release@v1’.</li>
<li>Use ‘<a href="https://hub.github.com/hub-release.1.html" rel="nofollow noopener">hub release edit</a>’ command to add the asset(your artifacts) to the release<br>
Code sample as below:</li>
</ol>
<pre><code class="lang-auto">      - name: Create Release          # create a new release v1
        id: create_release
        uses: actions/create-release@v1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          tag_name: v1
          release_name: Release v1
          draft: false
          prerelease: false
      - name: Attach binaries to release
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        run: |
          set -x
          assets=()
          for asset in ./bin/*; do               #loop the assets in the bin folder
            assets+=("-a" "$asset")
          done
          hub release edit "${assets[@]}" -m "Release v1" "v1"          # add the assets to the Release v1
</code></pre>
<p>Please check my answer in similar ticket <a href="https://github.community/t/release-assets-not-appearing/118148/2">here</a> for your reference.</p>
<p>You can use github actions to add assets as well, which you can find in marketplace <a href="https://github.com/marketplace?query=upload+asset+release" rel="nofollow noopener">here</a>.</p>
<p>Thanks.</p>

**Comment by thebino** [comment, marked as answer]
<p>Thanks, I didn’t knew that I can download ALL artifacts together.<br>
I solved it in a separate job and used a 3rd party action for uploads</p>
<pre><code class="lang-auto">    - name: Download artifact
      uses: actions/download-artifact@v2

    - name: Upload to Release
      uses: meeDamian/github-release@2.0
</code></pre>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/thebino">@thebino</a>,</p>
<p>Glad to know it’s resolved and thanks for sharing!</p>
<p>Happy coding!</p>
